### PR TITLE
Performance Profiler: weekly report subscribe page

### PR DESCRIPTION
**Pull request by gcsecsey**
Fixes https://github.com/Automattic/dotcom-forge/issues/8823

## Proposed Changes

* Add a new `/speed-test-tool/weekly-report` URL
* Make the site-build re-usable for the 404 page later (Iix1FFHIQdfECIXn5JKT5s-fi-4986_87188)
* On page load, send a request to the `/lead` endpoint to subscribe the user to weekly updates

## Why are these changes being made?

https://github.com/Automattic/dotcom-forge/issues/8823

### Logged-in
* Go to `/speed-test` and start a new test
* In the Calypso environment, copy over the finished test results eg. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzA1N30.7KXaBL1qcm0y0GqvM8ymJ0pbST9ls6P09_hri8EaKSI`
* Open the devtools, and go to the Network tab
* On the loaded test page, scroll down to the weekly report signup section, and click the Sign Up button
* Check that the site is redirected to the `/speed-test-tool/weekly-report` page
* On the Network tab, check that the POST request to the `/lead` endpoint is sent
* Test the loading and error state by blocking the `lead` request or modifying the flags in the `weekly-report/index.tsx` file

| Loading | Successful request | Error |
|--------|--------|--------|
| CleanShot 2024-09-02 at 12 49 23@2x | CleanShot 2024-09-02 at 12 48 41@2x | CleanShot 2024-09-02 at 12 49 54@2x | 

> [!NOTE]
> The log-in redirection can't be tested on the Calypso environment because of the environment mismatch when clicking the log-in link from an email. After deploying this PR, we will have to test the logged-out flow in production.

### Logged-out
* Go to `/speed-test` and start a new test
* In the Calypso environment, copy over the finished test results eg. `/speed-test-tool?url=https%3A%2F%2Fwordpress.com%2F&hash=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NzA1N30.7KXaBL1qcm0y0GqvM8ymJ0pbST9ls6P09_hri8EaKSI`
* Open the devtools, and go to the Network tab
* On the loaded test page, scroll down to the weekly report signup section, and click the Sign Up button
* Check that you're redirected to the `/log-in` page and that the `redirect_to` parameter preserves the query params
* Enter a test email address and create a new account
* Check that you're redirected to the success page and that the POST request to the `/lead` endpoint is sent

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by gavande1 — APPROVED** [review]
Great work. LGTM 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16630624

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @gcsecsey for including a screenshot in the description! This is really helpful for our translators.

File: client/performance-profiler/pages/weekly-report/index.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+					ctaHref={ `/speed-test-tool/weekly-report?url=${ url }&hash=${ hash }` }
+					secondaryMessage={ secondaryMessage }
+				/>
+			) }
+			{ isSuccess && (
+				<MessageDisplay
+					displayBadge
+					title={ translate( 'You’re all set!' ) }
+					message={ translate(
+						"We'll send you a weekly performance report for {{strong}}%s{{/strong}} so you can keep an eye on your site's speed. The first email is on it's way now.",
```
Comment (emilyaudela) [inline client/performance-profiler/pages/weekly-report/index.tsx:130]: small typo:
it's --> its

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16630624).

**Pull request merged**